client/client.go

```diff
@@ -333,6 +333,10 @@ var doNoTimeoutAndRetry = &doOptions{
 func (client *Client) do(method, path string, query url.Values, headers map[string]string, body io.Reader, v interface{}, opts *doOptions) (statusCode int, err error) {
 	opts = ensureDoOpts(opts)
 
+	if err := client.checkMaintenanceJSON(); err != nil {
```

Comment:
Should we only check this file in case we fail to talk to the socket?

Reply:
well eventually I think we should not even try the attempt to talk to the socket if we know via the maintenance.json file says that snapd is not around to talk, as some requests may have an unreasonably long timeout set by default, so it's a better user experience IMHO to get prompted quicker that something is refreshing than wait to try it and then be told that it's down for maintenance. 

towards that end, I would like to leave this at the top of the function